src/Microsoft.Health.Dicom.Core/Features/Query/QueryParser.cs

```diff
@@ -180,10 +201,15 @@ private bool TryParseDicomAttributeId(string attributeId, out DicomTag dicomTag)
             return succeed;
         }
 
-        private static void ValidateIfTagSupported(DicomTag dicomTag, string attributeId, QueryResource resourceType)
+        private static void ValidateIfTagSupported(DicomTag dicomTag, string attributeId, QueryResource resourceType, IDictionary<QueryResource, HashSet<DicomTag>> customTagMapping = null)
         {
             HashSet<DicomTag> supportedQueryTags = QueryLimit.QueryResourceTypeToTagsMapping[resourceType];
 
+            if (customTagMapping != null && customTagMapping.ContainsKey(resourceType))
+            {
+                supportedQueryTags.UnionWith(customTagMapping[resourceType]);
```

Comment:
nit: another way to write this is:
if (customTagMapping != null)
 {
                supportedQueryTags.UnionWith(customTagMapping.GetValueOrDefault(resourceType,new HashSet());
}

just personal preference to save some word. :)